erpnext/edi/doctype/common_code/common_code.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import hashlib
+
+import frappe
+from frappe import _
+from frappe.model.document import Document
+from frappe.utils.data import get_link_to_form
+from lxml import etree
+
+
+class CommonCode(Document):
+	# begin: auto-generated types
+	# This code is auto-generated. Do not modify anything in this block.
+
+	from typing import TYPE_CHECKING
+
+	if TYPE_CHECKING:
+		from frappe.core.doctype.dynamic_link.dynamic_link import DynamicLink
+		from frappe.types import DF
+
+		additional_data: DF.Code | None
+		applies_to: DF.Table[DynamicLink]
+		code_list: DF.Link
+		common_code: DF.Data
+		description: DF.SmallText | None
+		title: DF.Data
+	# end: auto-generated types
+
+	def before_validate(self):
+		if self.title and len(self.title) > 140:
+			self.title = self.title[:139] + "…"
+
+	def validate(self):
+		self.validate_distinct_references()
+
+	def validate_distinct_references(self):
+		"""Ensure no two Common Codes of the same Code List are linked to the same document."""
```

Comment:
For generating messages, it is very useful to have exactly one common code per ERPNext entity. However, for parsing messages, it would be useful to map multiple common codes to the same ERPNext entity. This is currently not possible due to this validation.

Reply:
@blaggacao any idea how to resolve this?

Reply:
I'd recommend binding in the context of an EDI Template. Within this narrowed down context, this restriction is likely more suitable, but also the binding list is usually short enough that no duplicates would typically happen.

But this restriction definitely, regardless, makes sense `on_update` within https://github.com/frappe/erpnext/pull/44174, specifically on [`bound_common_codes`](https://github.com/frappe/erpnext/pull/44174/files#diff-f3b34a58e2b8bf8d6892dfd41d9a5a4d076535000ad54d7ef12f669326128cd5R46)